crates/voicevox_core/src/internal.rs

```diff
@@ -184,10 +236,44 @@ impl Internal {
         phoneme_size: i64,
         f0: *const f32,
         phoneme: *const f32,
-        speaker_id: *const i64,
+        speaker_id: usize,
         output: *mut f32,
     ) -> Result<()> {
-        unimplemented!()
+        if !self.initialized {
+            return Err(Error::UninitializedStatus);
+        }
+
+        let status = self
+            .status_option
+            .as_mut()
+            .ok_or(Error::UninitializedStatus)?;
+
+        if !status.validate_speaker_id(speaker_id) {
+            return Err(Error::InvalidSpeakerId { speaker_id });
+        }
+
+        let (model_index, speaker_id) =
+            if let Some((model_index, speaker_id)) = get_model_index_and_speaker_id(speaker_id) {
+                (model_index, speaker_id)
+            } else {
+                return Err(Error::InvalidSpeakerId { speaker_id });
+            };
+
+        if model_index >= Status::MODELS_COUNT {
+            return Err(Error::InvalidModelIndex { model_index });
+        }
+
+        let f0_slice = unsafe { std::slice::from_raw_parts(f0, length as usize) };
+        let phoneme_slice = unsafe { std::slice::from_raw_parts(phoneme, phoneme_size as usize) };
+
+        let input_tensors = vec![ndarray::arr1(f0_slice), ndarray::arr1(phoneme_slice)];
+
+        let result = status.decode_session_run(model_index, input_tensors)?;
```

Comment:
C++ 実装では `f0` と `phoneme` を前後に太らせて音が途切れることに対処する workaround が入っていますが、まだそれは実装していません。

https://github.com/VOICEVOX/voicevox_core/blob/c47d05576e04360735c29de3e4fc73424440c48c/core/src/core.cpp#L408-L447

Reply:
念のためにワークアラウンドを入れるtodoコメントを書いて頂けると！！
```suggestion
        // TODO: 音が途切れてしまうのを避けるworkaround処理を入れる
        let f0_slice = unsafe { std::slice::from_raw_parts(f0, length as usize) };
        let phoneme_slice = unsafe { std::slice::from_raw_parts(phoneme, phoneme_size as usize) };

        let input_tensors = vec![ndarray::arr1(f0_slice), ndarray::arr1(phoneme_slice)];

        let result = status.decode_session_run(model_index, input_tensors)?;
```